numba/types/__init__.py

```diff
@@ -23,6 +23,8 @@
 Any = Phantom('any')
 undefined = Undefined('undefined')
 string = Opaque('str')
+code_type = Opaque('code')
+pyfunc_type = Opaque('pyfunc')
```

Comment:
Does this need to be `@typeof_impl.register`'d somewhere?

Reply:
`pyfunc_type` doesn't need registering since it is just used for output of `make_function` nodes in type inference. We don't have a use case other than this but we can add it later if necessary.